code/datums/gamemodes/crash.dm

```diff
@@ -0,0 +1,233 @@
+		CRASH("Shuttle [shuttle_id] wasn't found and can't be loaded")
+		return FALSE
+
+	var/datum/map_template/shuttle/S = SSmapping.shuttle_templates[shuttle_id]
+	var/obj/docking_port/stationary/L = SSshuttle.getDock("canterbury_loadingdock")
+	shuttle = SSshuttle.action_load(S, L)
+
+	// Reset all spawnpoints after spawning the ship
+	GLOB.marine_spawns_by_job = shuttle.marine_spawns_by_job
+	GLOB.jobspawn_overrides = list()
+	GLOB.latejoin = shuttle.spawnpoints
+	GLOB.latejoin_cryo = shuttle.spawnpoints
+	GLOB.latejoin_gateway = shuttle.spawnpoints
+
+	// Create xenos
+	var/number_of_xenos = length(xenomorphs)
+	var/datum/hive_status/normal/HN = GLOB.hive_datums[XENO_HIVE_NORMAL]
+	for(var/i in xenomorphs)
+		var/datum/mind/M = i
+		if(M.assigned_role == ROLE_XENO_QUEEN)
+			transform_ruler(M, number_of_xenos > HN.xenos_per_queen)
+		else
+			transform_xeno(M)
+
+
+/datum/game_mode/crash/post_setup()
+	. = ..()
+
+	// Reset all spawnpoints after spawning the ship
+	GLOB.marine_spawns_by_job = shuttle.marine_spawns_by_job
+	GLOB.jobspawn_overrides = list()
+	GLOB.latejoin = shuttle.spawnpoints
+	GLOB.latejoin_cryo = shuttle.spawnpoints
+	GLOB.latejoin_gateway = shuttle.spawnpoints
+
+	var/list/validdocks = list()
+	for(var/obj/docking_port/stationary/S in SSshuttle.stationary)
+		if(!shuttle.check_dock(S, silent=TRUE))
+			continue
+		validdocks += S.name
+
+	var/dock = pick(validdocks)
+
+	var/obj/docking_port/stationary/target
+	for(var/obj/docking_port/stationary/S in SSshuttle.stationary)
+		if(S.name == dock)
+			target = S
+			break
+
+	if(!target)
+		CRASH("Unable to get a valid shuttle target!")
+		return
+
+	addtimer(CALLBACK(src, .proc/crash_shuttle, target), 9 MINUTES) // TODO: REMOVE ADMIN TIMING
+	// addtimer(CALLBACK(src, .proc/crash_shuttle, target), rand(9 MINUTES, 11 MINUTES)) // TODO: FIX timing here
+
+
+/datum/game_mode/crash/announce()
+	to_chat(world, "<span class='round_header'>The current map is - [SSmapping.configs[GROUND_MAP].map_name]!</span>")
+	priority_announce("Scheduled for landing in T-10 Minutes. Prepare for landing. Known hostiles near LZ. Detonation Protocol Active, planet disposable. Marines disposable.", type = ANNOUNCEMENT_PRIORITY) // TODO: Better text.
+	playsound(shuttle, 'sound/machines/warning-buzzer.ogg', 75, 0, 30)
+	
+
+/datum/game_mode/crash/proc/add_larva()
+	var/datum/hive_status/normal/HS = GLOB.hive_datums[XENO_HIVE_NORMAL]
+	HS.stored_larva += 1
+
+
+/datum/game_mode/crash/proc/crash_shuttle(obj/docking_port/stationary/target)
+	shuttle.crashing = TRUE
+	SSshuttle.moveShuttleToDock(shuttle.id, target, FALSE) // FALSE = instant arrival
+	shuttle_landed = TRUE
+
+	announce_bioscans(TRUE, 0, FALSE, TRUE) // Announce exact information to the xenos.
+
+	addtimer(CALLBACK(src, .proc/add_larva), 5 MINUTES, TIMER_LOOP)
+	addtimer(CALLBACK(src, .proc/announce_bioscans, TRUE, 0, FALSE, TRUE), 5 MINUTES, TIMER_LOOP)
+
+
+/datum/game_mode/crash/check_finished()
```

Comment:
This should check if there are marines and no xenos, and the `CRASH_MINIMUM_TIME + SSticker.round_start_time` isn't over, and then spawn more xenos.